### ISSUE TITLE
perf(model): make V4.1 source and reader layers eligible for incremental decode

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13480]
# perf(model): make V4.1 source and reader layers eligible for incremental decode

```routing
repo: fak
lane: model
paths: ["internal/model/v41_forward.go", "internal/model/v41_incremental.go", "internal/model/v41_incremental_session_test.go"]
expected_steps: 5
priority: P1
```

## Working spine
Extend the existing one-token V4.1 forward to the published ratio-2 compressed and shared source/reader attention roles, with retained state and transactional rollback.

## Current state
#13311 and #13313 landed an incremental Session.Step seam for plain per-layer attention. But v41IncrementalEligible in internal/model/v41_forward.go:2714-2740 requires every layer to be PerLayer with ratio <= 1. The published V4.1 config admitted in internal/model/v41_config.go:345-364 requires ratio-2 and KV/index source layers. Thus that configuration cannot enter forwardV41Step; stepV41 falls to forwardV41 at v41_forward.go:2783, whose session path appends the token and recomputes its full history (:1955-1982). This is a structural gap, not a measured production throughput claim: the official safetensors artifact is currently load-refused. This leaf can use the published-config software fixture now; production use depends on the #12640 load gate.

## Why this is next
An incremental path limited to a plain attention shape cannot serve the architecture the V4.1 effort targets. Once loading is normalized, each generated token would replay the growing prefix until this role-specific state is retained.

## Parent context
#12640 owns full V4.1 execution and physical qualification. #13311/#13313 cover the plain-layer incremental seam; #13323 covers ratio-2 full-forward numeric behavior; #13346 covers suffix Prefill. This leaf owns source/reader continuation in Step only. The published-config software fixture is executable before the official artifact loads. Keep the live production-use qualification with #12640; do not use issue closure alone as an execution witness.

## Core through-line
Seed retained compressor/source/index state from Prefill for the published 40-layer role plan, advance only one new position per layer in forwardV41Step, and remove the role-based eligibility refusal only when that state is valid. Keep full-forward as an explicit reference/fallback for unsupported states.

## Gold-plating boundary
No safetensors loader implementation, GGUF conversion, new attention kernel, suffix Prefill optimization, physical 30 tok/s claim, or weakened numerical tolerance.

## Repro witness
From public fak: ```rg -n -C 2 'plan.Role != V41AttentionRolePerLayer|forwardV41\(\[\]int\{id\}' internal/model/v41_forward.go```; inspect the published compression schedule in internal/model/v41_config.go:345-364. The current eligibility guard rejects the published plan before the incremental branch.

## Done condition / witness
A pinned published 40-layer schedule takes the incremental Session.Step path for 128 generated steps, processes one new position per layer per step independent of prefix length, matches the full-forward reference under existing strict tolerances, and rolls back failed steps. Witness: ```go test -count=1 ./internal/model -run '^TestV41PublishedRolesIncrementalStep'```.

## Verifiable Witness
```text
go test -count=1 ./internal/model -run '^TestV41PublishedRolesIncrementalStep'
go test -count=1 ./internal/model
```

## Definition of done
- [ ] Published ratio-2 compressor/source/reader/index roles have retained state seeded by Prefill and consumed by Step.
- [ ] A 32- and 128-token-prefix counter witness shows exactly one new position per layer per Step; no full-history replay or prefix-sized state clone.
- [ ] Full-vocabulary logits and retained state match the full-forward oracle for fixed token sequences under predeclared tolerances.
- [ ] An injected late-layer failure rolls back all role state; retry matches the reference; named and package tests pass.

## File:Line Seam
internal/model/v41_forward.go:1950-1982,2714-2788; v41_config.go:345-364; existing incremental composition in v41_incremental.go.

## Blast radius and fallback
Primary lane model, V4.1 text decode only. Keep the current full-forward reference for unsupported or unseeded states and maintain typed refusal on malformed plans. This is [SW-VERIFIED] until a separately leased physical request uses the role path; no hardware rate is asserted here.

## Likely files
- internal/model/v41_forward.go
- internal/model/v41_incremental.go
- internal/model/v41_incremental_session_test.go

## Acceptance gate
The named test invokes Session.Step on the published role schedule; counters, parity and rollback pass. A helper-only test or no-tests-to-run output fails acceptance.

## Closure binding
Resolving signed commit cites this issue and carries (fak model).

## Value
- Centrality: Core
- P1 Context: advanced - incremental path reaches target V4.1 attention roles
- P2 Net value: advanced - removes prefix replay per generated token
- P3 Adaptation: preserved - current full-forward oracle and typed fallback
- P4 Operations: advanced - route counters distinguish incremental from fallback

## Work estimate
Estimate: 5 points

## Overall completion contribution
Contribution: 5/8 points

## Completion standard
development
## Witness
The named published-role regression must execute Session.Step, prove the incremental probe fires, and count one new position per layer for each token. Run go test -count=1 ./internal/model -run '^TestV41PublishedRolesIncrementalStep' and go test -count=1 ./internal/model.

**Comment by anthony-chaudhary** [comment]
Resolved in commit c48f844520 (on origin/main). Verification witness: \go test -count=1 ./internal/model -run '^TestV41PublishedRolesIncrementalStep'\ -> ok (4 subtests: source step matches full-sequence reference; source seeded mid-group continues without replaying the prefix; shared registry is the cross-layer reader view; retained state stays bounded across a long source run). Package witness: \go test -count=1 ./internal/model -skip TestExpertResidencyLFUDecayBeatsLRUOnRoutingJitter\ -> ok; \go vet ./internal/model\ clean; gofmt clean; 5-gate boundary \ak-boundary check --all\ PASS (0 violations). The landing commit used 'Ref #13480'; the change is [SW-VERIFIED] and already on public origin/main.
